packages/transport/src/transports/udp.ts

```diff
@@ -30,12 +31,27 @@ export class UdpTransport extends AbstractApiTransport {
             sessionsClient,
             signal,
         });
+    }
 
+    public listen() {
         const enumerateRecursive = () => {
-            setTimeout(() => {
+            if (!this.listening) return;
+
+            this.enumerateTimeout = setTimeout(() => {
                 this.enumerate().promise.finally(enumerateRecursive);
             }, 500);
         };
         enumerateRecursive();
+
+        return super.listen();
+    }
```

Comment:
just a side note - I will probably need to do something like this also for usb. unfortunatelly, onconnect event in windows does not return serialNumber which we rely on. 

so my rough plan for now is
- on api level, introduce abstract listen member. its task will be to make api start emitting events. 
- for udp, it would be this code here
- for usb, it would be probably listening to onconnect events and doing enumeration after they are detected.